### PR TITLE
Adding Bicep file and updating Readme and Metadata files

### PR DESCRIPTION
**Pull request by msjasteppe**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Added Bicep file.
* Updated Readme file.
* Updated Metadata file.

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@msjasteppe - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msjasteppe - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by msjasteppe** [comment]
#sign-off

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  properties: {
+    zoneRedundant: true
+    isAutoInflateEnabled: true
+    maximumThroughputUnits: 8
+    kafkaEnabled: false
+  }
+}
+
+resource eventHub 'Microsoft.EventHub/namespaces/eventhubs@2021-11-01' = {
+  name: eventHubFullName
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:72]: use the `parent` property and remove `dependsOn` below (for all child resources)

**Comment by azure-quickstarts** [comment]
@msjasteppe - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+resource EventHubDataReceiver 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: eventHub
+  name: guid('${resourceGroup().id}-EventHubDataReceiver')
+  properties: {
+    roleDefinitionId: eventHubReceiverRoleId
+    principalId: reference(iotConnectorResourceId, '2022-05-15', 'Full').identity.principalId
+    principalType: 'ServicePrincipal'
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:178]: remove `dependsOn`

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+var eventHubName = 'devicedata'
+var eventhubNamespaceName = 'en-${basename}'
+var eventHubFullName = '${eventhubNamespaceName}/${eventHubName}'
+var eventHubAuthRuleName = '${eventHubFullName}/devicedatasender'
+var workspaceName = replace('hw-${basename}', '-', '')
+var fhirServiceName = '${workspaceName}/fs-${basename}'
+var iotConnectorName = '${workspaceName}/hi-${basename}'
+var fhirDestinationName = '${iotConnectorName}/hd-${basename}'
+var fhirServiceResourceId = resourceId('Microsoft.HealthcareApis/workspaces/fhirservices', workspaceName, split(fhirServiceName, '/')[1])
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:50]: use the bicep symbolic reference for your resourceId e.g. `fhirService.id` and remove the vars

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  }
+  dependsOn: [
+    workspace
+    iotConnector
+  ]
+}
+
+resource FhirWriter 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: fhirService
+  name: guid('${resourceGroup().id}-FhirWriter')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:159]: use:
```
name: guid(fhirWriterRoleId, iotConnector.id, fhirService.id)
```

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    principalType: 'ServicePrincipal'
+  }
+  dependsOn: [
+    iotConnector
+  ]
+}
+
+resource EventHubDataReceiver 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: eventHub
+  name: guid('${resourceGroup().id}-EventHubDataReceiver')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:172]: use:
```
name: guid(eventHubReceiverRoleId, iotConnector.id, eventHub.id)
```

File: quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+resource FhirWriter 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: fhirService
+  name: guid('${resourceGroup().id}-FhirWriter')
+  properties: {
+    roleDefinitionId: fhirWriterRoleId
+    principalId: reference(iotConnectorResourceId, '2022-05-15', 'Full').identity.principalId
+    principalType: 'ServicePrincipal'
+  }
+  dependsOn: [
```
Comment (bmoore-msft) [inline quickstarts/microsoft.healthcareapis/workspaces/iotconnectors/main.bicep:165]: remove `dependsOn` - you won't need it once you change to symbolic references in the id

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@msjasteppe - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@msjasteppe - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by msjasteppe** [comment]
#sign-off

**Pull request merged**